src/app/products/products.component.html

```diff
@@ -1 +1,19 @@
-<p>products works!</p>
+<div class="row">
+  <ng-container
+    *ngFor="let product of productsObservable | async; let i = index"
+  >
+    <div class="col">
+      <div class="card" style="width: 15rem;">
+        <img [src]="product.imageUrl" class="card-img-top" alt="..." />
+        <div class="card-body">
+          <h5 class="card-title">{{ product.title }}</h5>
+          <p class="card-text">
+            {{ product.price | currency: 'USD' }}
+          </p>
+          <a href="#" class="btn btn-primary">Add to Cart</a>
+        </div>
+      </div>
+    </div>
+    <div *ngIf="(i + 1) % 2 === 0" class="w-100"></div>
```

Comment:
move logic out of html. Logic is easily tested when it lives in a component or service.